pkg/backend/endpoint_resolver.go

```diff
@@ -27,12 +28,17 @@ var ErrNotFound = errors.New("backend not found")
 type EndpointResolver interface {
 	// ResolvePodEndpoints will resolve endpoints backed by pods directly.
 	// returns resolved podEndpoints and whether there are unready endpoints that can potentially turn ready in future reconciles.
-	ResolvePodEndpoints(ctx context.Context, svcKey types.NamespacedName, port intstr.IntOrString,
-		opts ...EndpointResolveOption) ([]PodEndpoint, bool, error)
+	ResolvePodEndpoints(ctx context.Context, svckey types.NamespacedName, svc *corev1.Service, port intstr.IntOrString, opts ...EndpointResolveOption) ([]IpEndpoint, bool, error)
```

Comment:
There's no need for a breaking API change here. The variadic list of `EndpointResolveOption` argument should eliminate that necessity, can just create a new one to the effect of `WithService(*corev1.Service)`. This package is currently at v2, merging this would require tagging a v3 and that creates more churn for adoption than necessary.